pkg/execution/state/redis_state/queue.go

```diff
@@ -1382,29 +1398,48 @@ func (q *queue) Peek(ctx context.Context, partition *QueuePartition, until time.
 		args,
 	).ToAny()
 	if err != nil {
-		return nil, fmt.Errorf("error peeking queue items: %w", err)
+		return nil, fmt.Errorf("error peeking queue items (limit: %d): %w", limit, err)
 	}
 
 	returnedSet, ok := peekRet.([]any)
 	if !ok {
 		return nil, fmt.Errorf("unknown return type from peek: %T", peekRet)
 	}
 
-	var items, missingQueueItems []any
+	var potentiallyMissingItems, allQueueItemIds []any
 	if len(returnedSet) == 2 {
-		items, ok = returnedSet[0].([]any)
+		potentiallyMissingItems, ok = returnedSet[0].([]any)
 		if !ok {
 			return nil, fmt.Errorf("unexpected first item in set returned from peek: %T", peekRet)
 		}
 
-		missingQueueItems, ok = returnedSet[1].([]any)
+		allQueueItemIds, ok = returnedSet[1].([]any)
 		if !ok {
 			return nil, fmt.Errorf("unexpected first item in set returned from peek: %T", peekRet)
 		}
 	} else if len(returnedSet) != 0 {
 		return nil, fmt.Errorf("expected zero or two items in set returned by peek: %v", returnedSet)
 	}
 
+	items := make([]any, 0, len(allQueueItemIds))
```

Comment:
This finds and gracefully handles missing items. We moved this out of Lua and into the executor to speed up script processing.